File: Apis/Domain/Entities/Task.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class Task : BaseEntity
{
    public DateTime? EndDate { get; set; }

    public int Id { get; set; }

    public DateTime? StartDate { get; set; }

    public int? Teamid { get; set; }

    public string? Priority { get; set; }

    public string? Status { get; set; }

    public string? Assignee { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public virtual Team? Team { get; set; }
}

File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/Task.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Task")
public class Task {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Lob
    @Column(name = "description")
    private String description;

    @Size(max = 100)
    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Size(max = 100)
    @Column(name = "assignee", length = 100)
    private String assignee;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Size(max = 50)
    @Column(name = "status", length = 50)
    private String status;

    @Size(max = 50)
    @Column(name = "priority", length = 50)
    private String priority;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teamid")
    private Team teamid;

}